src/prpy/util.py

```diff
@@ -693,3 +693,44 @@ def FindCatkinResource(package, relative_path):
     else:
         raise IOError('Loading resource "{:s}" failed.'.format(
                       relative_path))
+
+
+def IsAtTrajectoryStart(robot, trajectory):
+    """
+    Check if robot's active DOFs match the start configuration of a trajectory.
+
+    Examines the current active DOF values of the specified robot and compares
+    these values to the first waypoint of the specified trajectory.  If every
+    DOF value differs by less than the DOF resolution of the specified
+    joint/axis, then the function returns True.  Otherwise, it returns False.
+
+    @param robot: the robot whose active DOFs will be checked
+    @param trajectory: the trajectory whose start configuration will be checked
+    @returns: True if the robot's active DOFs match the given trajectory
+              False if one or more active DOFs differ by DOF resolution
+    """
+    # Get current active configuration of robot.
+    dof_indices = robot.GetActiveDOFIndices()
+    dof_values = robot.GetActiveDOFValues()
+    dof_resolutions = robot.GetActiveDOFResolutions()
```

Comment:
I still think that `ExecuteTrajectory` should ignore the active DOFs. Here are three concrete reasons:

1. **Backwards Compatibility.** `ExecuteTrajectory` has always ignored the active DOFs.
2. **OpenRAVE Convention.** `ExecuteTrajectory` performs the same function as `SetPath`, which ignores active DOFs.
3. **Safety.** The trajectory formed by an arbitrary subset of the trajectory's DOFs may not be collision free.

Honoring the active DOFs also leads to counterintuitive behavior when mixed with robot state savers. Depending upon the active DOFs the following code could: (1) execute a right arm trajectory, (2) execute a trajectory on a subset of the right arm's DOFs, or (3) do nothing. See below:

```python
traj = robot.right_arm.PlanToNamedConfiguration('home')
robot.ExecutePath(traj)
```

Finally, I think active DOFs and manipulators are a generally bad idea. The "active" concept introduce additional state that is a constant headache to manage. I'd like to entirely move PrPy away from active DOFs in the future. We're already considering doing so in the planning pipeline, because the concept of an "active manipulator" breaks on bimanual tasks.